src/compiler/checker.ts

```diff
@@ -2913,6 +2915,22 @@ namespace ts {
             }
         }
 
+        function baseTypeHasUnappliedOuterTypeParameters(type: Type): boolean {
+            const outerTypeParameters = (<InterfaceType>type).outerTypeParameters;
+            const typeArguments = (<TypeReference>type).typeArguments;
+            if (outerTypeParameters) {
+                // an unapplied type parameter is one 
+                // whose argument symbol is still the same as the parameter symbol
+                const numParameters = outerTypeParameters.length;
+                for (let i = 0; i < numParameters; i++) {

```

Comment:
Actually, if it's the case that type parameters are always applied from earlier indices onward, you could just check at the last index.